File: dasQt/utools/readh5.py
import numpy as np
import h5py

def readh5(filename):
    with h5py.File(filename, 'r') as f:
        for a in f:
            StrainRate = f[f'/{a}/Source1/Zone1/StrainRate'][:]
            spacing = f[f'/{a}/Source1/Zone1'].attrs['Spacing']


    #self.data = StrainRate.reshape(-1, StrainRate.shape[-1])
    dx          = spacing[0]
    dt          = spacing[1] * 1e-3
    nb, nt, nx  = StrainRate.shape
    data        = StrainRate[:, nt//2:, :].reshape(-1, nx)
    nt, nx      = data.shape

    return data, dx, dt, nt, nx

File: utools/denoise/readH5.m
function [data, dx, dt, nt, nx] = readH5(filename)
    info = h5info(filename);
    for k = 1:length(info.Groups)
        group_name = info.Groups(k).Name;
        strainRatePath = strcat(group_name, '/Source1/Zone1/StrainRate');
        spacingAttrPath = strcat(group_name, '/Source1/Zone1');
        
        StrainRate = h5read(filename, strainRatePath);
        spacing = h5readatt(filename, spacingAttrPath, 'Spacing');
    end

    dx = spacing(1);
    dt = spacing(2) * 1e-3;
    [nb, nt, nx] = size(StrainRate);
    data = reshape(StrainRate(:, floor(nt/2)+1:end, :), [], nx);
    [nt, nx] = size(data);

    return
end
